### PR TITLE
Removing old federation ci jobs

### PR DESCRIPTION
**Pull request by nikhiljindal**
Ref https://groups.google.com/forum/#!msg/kubernetes-sig-federation/SNmYeMjiduU/jSZ3rynvAgAJ

Removing old sig federation ci jobs. Also removing gci-gce-federation since node image does not impact federation.

cc @kubernetes/sig-federation-pr-reviews @madhusudancs

### CODE REVIEW

**Comment by krzyzacy** [comment]
/assign
you want to also delete corresponding `.env` files from jobs/ dir, and remove testgroup from the testgrid config

**Comment by nikhiljindal** [comment]
Thanks for the quick review @krzyzacy 
Updated and bazel test pass now :)

File: testgrid/config/config.yaml [hunk 1]
```diff
@@ -188,12 +188,8 @@ test_groups:
...
 - name: kubernetes-e2e-gce-federation-serial
   gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-federation-serial
-- name: kubernetes-e2e-gci-gce-federation
-  gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-federation
-- name: kubernetes-e2e-gce-federation-release-1.4
-  gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-federation-release-1.4
-- name: kubernetes-e2e-gce-federation-release-1.5
-  gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-federation-release-1.5
+- name: kubernetes-e2e-gce-federation-release-1.6
+  gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-federation-release-1.6
```
Comment (krzyzacy) [inline testgrid/config/config.yaml]: ahh, there's already a testgroup named `ci-kubernetes-e2e-gce-federation-release-1.6`, keep this or the other.

the name is not very consistent here :-|

Comment (nikhiljindal) [inline testgrid/config/config.yaml]: oh done

File: testgrid/config/config.yaml [hunk 2]
```diff
@@ -188,12 +188,8 @@ test_groups:
...
 - name: kubernetes-e2e-gce-federation-serial
   gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-federation-serial
-- name: kubernetes-e2e-gci-gce-federation
-  gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-federation
-- name: kubernetes-e2e-gce-federation-release-1.4
-  gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-federation-release-1.4
-- name: kubernetes-e2e-gce-federation-release-1.5
-  gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-federation-release-1.5
+- name: kubernetes-e2e-gce-federation-release-1.6
+  gcs_prefix: kubernetes-jenkins/logs/kubernetes-e2e-gce-federation-release-1.6
```
Comment (krzyzacy) [inline testgrid/config/config.yaml]: I mean there's also https://github.com/nikhiljindal/test-infra/blob/442ca49956d05bac41635612920c3b11890ef3bb/testgrid/config/config.yaml#L675-L676

Comment (krzyzacy) [inline testgrid/config/config.yaml]: I'll do a clean up pass, just don't add the entry here, and use `ci-kubernetes-e2e-gce-federation-release-1.6` for now.

Comment (nikhiljindal) [inline testgrid/config/config.yaml]: done

**Comment by nikhiljindal** [comment]
Thanks @krzyzacy Updated code as per comments.
PTAL

**Comment by krzyzacy** [comment]
/lgtm

do you have merge here?

**Comment by nikhiljindal** [comment]
No. I dont have write access

**Pull request merged**